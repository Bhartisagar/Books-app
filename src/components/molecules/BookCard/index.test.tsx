import renderer from 'react-test-renderer';
import {  render, screen } from "@testing-library/react";
import Cards from ".";
import { fireEvent } from '@storybook/testing-library';


const bookData={
    "id": 1,
    "imgSrc": "card1.jpg",
    "imgAlt": "Bring Your Human to Work",
    "name": "Bring Your Human to Work",
    "author": "Erica Keswin",
    "readTime": "13-minute read",
    "readCount": "19.1k reads",
    "state": "CR",
    "category": "",
    "audio": true,
    "justAdded": false,
    "isTrending": false
}
const btnText="Finished";

test("first snapshot",()=>{
    const component = renderer.create(
        <Cards bookData={bookData} btnText={btnText} />
    )
    let tree=component.toJSON();

    expect(tree).toMatchSnapshot();
})

test('should render image',()=>{    
    render (<Cards bookData={bookData} btnText={btnText} />);
    const ImageDis = screen.getByRole("img")
    expect(ImageDis).toBeInTheDocument();
})

test('should render Book name',()=>{    
    render (<Cards bookData={bookData} btnText={btnText} />);
    const cardheading = screen.getByRole("heading",{name:"Bring Your Human to Work"})
    expect(cardheading).toBeInTheDocument();
})
test('should render Book author name',()=>{    
    render (<Cards bookData={bookData} btnText={btnText} />);
    const bookAuthor = screen.getByRole("heading",{name:"Erica Keswin"})
    expect(bookAuthor).toBeInTheDocument();
})
test('should render Book read time',()=>{    
    render (<Cards bookData={bookData} btnText={btnText} />);
    const bookAuthor = screen.getByRole("heading",{name:"13-minute read"})
    expect(bookAuthor).toBeInTheDocument();
})
test('should render Button',()=>{    
    render (<Cards bookData={bookData} btnText={btnText} />);
    const bookButton = screen.getByRole("button",{name:"Finished"})
    expect(bookButton).toBeInTheDocument();
})
test('should not render Button',()=>{    
    render (<Cards bookData={bookData} btnText="" />);
    const bookAuthor = screen.getByRole("heading",{name:"13-minute read"})
    expect(bookAuthor).toBeInTheDocument();
})

test('should render mapp data button',async()=>{
    render(<Cards bookData={bookData} btnText={btnText} />)
    const card1 = await screen.findByTestId("button-1")
    expect(card1).toBeInTheDocument();
})

const bookFunction = jest.fn();

test('should render mapp data',async()=>{
    render(<Cards bookData={bookData} btnText={btnText} />)
    const card2 = await screen.findByTestId("headings-1")
    expect(card2).toBeInTheDocument();
})
test('should render function',async()=>{
    render(<Cards bookData={bookData} btnText={btnText} bookFunction={bookFunction}/>)
    const card2 = await screen.findByTestId("headings-1")
    expect(card2).toBeInTheDocument();
})
   const fun = (bookData)=>{

   }
test('should render mapp data button fun',async()=>{
    render(<Cards bookData={bookData} btnText={btnText} bookFunction={fun}/>)
    const card1 = await screen.findByTestId("button-1")
    fireEvent.click(card1)
})
