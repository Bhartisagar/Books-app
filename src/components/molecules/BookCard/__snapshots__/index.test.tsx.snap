// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-cardCon-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="cards1"
>
  <img
    alt="Bring Your Human to Work"
    height={342}
    src="card1.jpg"
    width={340}
  />
  <div
    className="MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root makeStyles-cardcontent-4 MuiTypography-subtitle2"
      data-testid="headings-1"
    >
      Bring Your Human to Work
    </h6>
    <p
      className="MuiTypography-root makeStyles-cardcontent-4 MuiTypography-body1"
    >
      Erica Keswin
    </p>
    <div
      className="makeStyles-cardcontent-4"
      style={
        Object {
          "display": "inline-flex",
          "gap": "2px",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
        <path
          d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
        />
      </svg>
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        13-minute read
      </span>
    </div>
    <div
      style={
        Object {
          "display": "inline-flex",
          "gap": "2px",
          "marginLeft": "70px",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 9c2.7 0 5.8 1.29 6 2v1H6v-.99c.2-.72 3.3-2.01 6-2.01m0-11C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <span
        className="MuiTypography-root MuiTypography-caption"
        data-testid="readcount-1"
      >
        19.1k reads
      </span>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <div>
       
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-tabButtons-3"
        data-testid="button-1"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Finished
        </span>
      </button>
      <img
        alt="rantangle"
        className="makeStyles-rectangle-5"
        src="rectangle.svg"
      />
    </div>
    <div />
  </div>
</div>
`;
